internal/session/testing.go

```diff
@@ -273,3 +273,31 @@ func TestTofu(t testing.TB) []byte {
 func TestCert(jobId string) (ed25519.PrivateKey, []byte, error) {
 	return newCert(context.Background(), jobId, []string{"127.0.0.1", "localhost"}, time.Now().Add(5*time.Minute), rand.Reader)
 }
+
+// TestListSessions exposes the repo listSessions method for testing purposes.
+func TestListSessions(t testing.TB, repo *Repository, ctx context.Context, opts ...Option) ([]*Session, time.Time) {
+	targets, ttime, err := repo.listSessions(ctx, opts...)
+	require.NoError(t, err)
+	return targets, ttime
+}
```

Comment:
Do we need these functions? There's no risk of import cycles in this hierarchy. We can just have an internal test file that tests the unexported methods?